NuRadioReco/utilities/trace_utilities.py

```diff
@@ -111,3 +112,53 @@ def get_electric_field_energy_fluence(electric_field_trace, times, signal_window
         f_signal -= f_noise * np.sum(signal_window_mask) / np.sum(noise_window_mask)
 
     return f_signal * dt * conversion_factor_integrated_signal
+
+def upsampling_fir(trace, original_sampling_frequency, int_factor=2, ntaps=2**7):
+    """
+    This function performs an upsampling by inserting a number of zeroes
+    between samples and then applying a finite impulse response (FIR) filter.
+
+    Parameters
+    ----------
+    trace: array of floats
+        Trace to be upsampled
+    original_sampling_frequency: float
+        Sampling frequency of the input trace
+    int_factor: integer
+        Upsampling factor. The resulting trace will have a sampling frequency
+        int_factor times higher than the original one
+    ntaps: integer
+        Number of taps (order) of the FIR filter
+
+    Returns
+    -------
+    upsampled_trace: array of floats
+        The upsampled trace
+    """
+
+    if (np.abs(int(int_factor) - int_factor) > 1e-3):
+        warning_msg  = "The input upsampling factor does not seem to be close to an integer."
+        warning_msg += "It has been rounded to {}".format(int(int_factor))
+        logger.warning(warning_msg)
+
+    int_factor = int(int_factor)
+
+    if (int_factor <= 1):
+        error_msg = "Upsampling factor is less or equal to 1. Upsampling will not be performed."
+        raise ValueError(error_msg)
+
+    n_zeroes = int_factor - 1
+    zeroed_trace = []
```

Comment:
This is fine for now but if performance matters later, it would be much better to make a zeroed trace first and fill in the non-zero samples (so you don't have so many needless memory allocations).

Reply:
You're right. I can change that.

Reply:
Changed.